bot/siteMonitor.py

```diff
@@ -0,0 +1,98 @@
+import urllib.request
+import difflib
+import time
+from bs4 import BeautifulSoup
+import re
+
+request_time = 10
+    
+    
+    
+#code for diff
+def diff(first, second):
+        
+        item_list = []
+        
+        second = set(second)
+        
+        for item in first:
+            if item not in second:
+                item_list.append(item)
+        
+        return item_list
+
+
+
+def takeDifferences(lists, i):
+    
+    
+    for new, old in zip(lists[i+1], lists[i]):
+        
+        differences = diff(new, old)
+        
+        #not empty list
+        if(differences != []):
+            if len(differences) < 2:
+                print(differences)
+            else:
+                print("Lot of news pal!")    
+        else:
+           # print("We have to go deeper")
+            if i < len(lists)-3:
+                i += 2
+                takeDifferences(lists, i)
+            else:
+                break    
+            #save previous changes
+
+    
+#save site content    
+url = "https://www.wykop.pl/wykopalisko/najnowsze/"
+html = urllib.request.urlopen(url).read()
+soup = BeautifulSoup(html, "html.parser")
+
+#repeat
+while(True):
+    
+    time.sleep(request_time)
+    url2 = "https://www.wykop.pl/wykopalisko/najnowsze/"
+    html2 = urllib.request.urlopen(url2).read()
+    soup2 = BeautifulSoup(html2, "html.parser")
+    
+    #region for lists of tags
+    lists = []
+    
+    #lists.append(soup.find_all([ 'h1']))
+    #lists.append(soup2.find_all([ 'h1']))
+ 
+     
+    lists.append(soup.find_all([ 'h2']))
```

Comment:
I believe that instead of appending these tags one by one, you can for example achieve the same by list comprehension. How about something like that:

`tags = ['h1', 'h2', 'h3', 'h4', 'h5', 'p']`
`lists = [soup.find_all(tag) for tag in tags]`

Reply:
I also wonder if we need to call `find_all` from `BeautifulSoup` so many times. If we could call find_all once using all of these tags and making sure of their proper order in our list without too much additional cost, it would be great.